src/container/RelayRenderer.js

```diff
@@ -342,12 +281,53 @@ class RelayRenderer extends React.Component {
     this.mounted = false;
   }
 
+  /**
+   * @private
+   */
+  _resolveContainerProps(): Object {
+    if (!this.containerProps) {
+      const {environment, queryConfig} = this.props;
+      this.containerProps = {
+        ...queryConfig.params,
+        ...mapObject(
+          this.querySet,
+          query => createFragmentPointerForRoot(environment, query)
+        ),
+      };
+    }
+    return this.containerProps;
+  }
+
+  /**
+   * @private
+   */
+  _calculateRenderArgs(): RelayRendererRenderArgs {
+    const {readyState} = this.state;
+    if (readyState) {
+      return {
+        done: readyState.done,
+        error: readyState.error,
+        props: readyState.ready ? this._resolveContainerProps() : null,
+        retry: this._retry.bind(this),
+        stale: readyState.stale,
+      };
+    } else {
+      return {
+        done: false,
+        error: null,
+        props: null,
+        retry: this._retry.bind(this),
+        stale: false,
+      };
+    }
+  }
+
   render(): ?React$Element {
     let children;
     let shouldUpdate = this._shouldUpdate();
     if (shouldUpdate) {
       const {Container, render} = this.props;
-      const {renderArgs} = this.state;
+      const renderArgs = this._calculateRenderArgs();
```

Comment:
Nice. I like how you changed the `renderArgs` (and `containerProps`) to be lazily computed.